File: src/shared-services/rest/laureate-service-sample-03/laureate.export/src/main/java/de/htwberlin/lla/master/towardseda/sample03/laureateservice/laureate/export/model/Laureate.java
package de.htwberlin.lla.master.towardseda.sample03.laureateservice.laureate.export.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.*;

import java.time.LocalDate;

@Data
@AllArgsConstructor(staticName = "of")
@NoArgsConstructor
@ToString
@Builder
public class Laureate {

    @NonNull
    private int id;

    @NonNull
    private String name;

    private String givenName;

    @NonNull
    private String gender;

    @NonNull
    private String wikiUrl;

    @NonNull
    private String laureateUrl;

    @NonNull
    private String externalUrl;

    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate birth;

    private Integer birthYear;

    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate death;

    private Integer deathYear;

    private String birthLocation;

    private String deathLocation;
}

File: src/frontend/src/app/model/laureate.ts
export class Laureate {
    constructor(
        readonly id: number,
        readonly name: string,
        readonly givenName: string,
        readonly gender: string,
        readonly wikiUrl: string,
        readonly laureateUrl: string,
        readonly externalUrl: string,
        readonly birth: Date,
        readonly birthYear: number,
        readonly death: Date,
        readonly deathYear: number,
        readonly birthLocation: string,
        readonly deathLocation: string,
        readonly prizesErr?: string,
        readonly prizes?: {
            category: string;
            laureate: number;
            year: number;
            prizeAmount: number;
            prizeAmountAdjusted: number;
        }[]
    ) {}
}
